bytecomp/bytelink.ml

```diff
@@ -90,16 +97,11 @@ let add_ccobjs origin l =
 
 let missing_globals = ref Ident.Map.empty
 let provided_globals = ref Ident.Set.empty
-let badly_ordered_dependencies : (string * string) list ref = ref []
-
-let add_unique x xs =
-  if List.mem x xs then xs else x::xs
+let badly_ordered_dependencies : DepSet.t ref = ref DepSet.empty
 
 let record_badly_ordered_dependencies (id, compunit) =
```

Comment:
Nitpick: I think this should be `record_badly_ordered_dependency` (we record just one dependency)